### PR TITLE
Use C++17 Convention Methods in PyTorch

### PR DESCRIPTION
**Pull request by fffrog**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137958

Detailed Descriptions:
- `std::is_same<X, Y>::value` -> `std::is_same_v<X, Y>`
- `std::enable_if<C, T>::type` -> `std::enable_if_t<C, T>`
- and so on

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @EikanWang

### CODE REVIEW

**Review by janeyx99 — APPROVED** [review]
Please fix lint, change LGTM otherwise.

The line length ones made the change a bit harder to review, so would prefer not having those be mixed in next time

File: aten/src/ATen/native/cpu/FusedAdagradKernel.cpp [hunk 1]
```diff
@@ -12,52 +12,54 @@ namespace at::native {
...
+        const double weight_decay,
+        const bool maximize,
+        const float* grad_scale_ptr,
+        int64_t size) {
   using lpVec = at::vec::Vectorized<scalar_t>;
   using fVec = at::vec::Vectorized<opmath_t>;
   int64_t d = 0;
   for (; d < size - (size % lpVec::size()); d += lpVec::size()) {
     lpVec param_lpvec = lpVec::loadu(param_ptr + d);
-    auto [param_vec1, param_vec2] = vec::convert_to_float<scalar_t>(param_lpvec);
```
[the remark this answers is not in the document]
Comment (FFFrog) [reply aten/src/ATen/native/cpu/FusedAdagradKernel.cpp:34]: I formatted this function manually, and in the latest commit I reverted it :D

**Comment by FFFrog** [comment]
> Please fix lint, change LGTM otherwise.
> 
> The line length ones made the change a bit harder to review, so would prefer not having those be mixed in next time

Thanks your for your comments, have got it now :D

**Pull request closed without merging**